test_util/aws.py

```diff
@@ -58,13 +59,22 @@ def __init__(self, stack_name, boto_wrapper):
         self.boto_wrapper = boto_wrapper
         self.stack = self.boto_wrapper.resource('cloudformation').Stack(stack_name)
 
-    def wait_for_stack_creation(self):
-        self.boto_wrapper.client('cloudformation').get_waiter('stack_create_complete').\
-            wait(StackName=self.stack.stack_name)
+    def boto_waiter(self, client, waiter, **kwargs):
+        """ Waiter error messages are unhelpful, so pump out stack event infomation
 
-    def wait_for_stack_deletion(self):
-        self.boto_wrapper.client('cloudformation').get_waiter('stack_delete_complete').\
-            wait(StackName=self.stack.stack_name)
+        See http://boto3.readthedocs.io/en/latest/reference/services/index.html
+        under CloudFormation and EC2 client Waiter for common options
+
+        NOTE: Cache stack ID before starting so stacks that failed to create can be examined
+        """
+        stack_id = self.stack.stack_id
+        cf_client = self.boto_wrapper.client(client)
+        try:
+            cf_client.get_waiter(waiter).wait(**kwargs)
+        except botocore.exceptions.WaiterError:
+            stack_info = repr(self.boto_wrapper.client('cloudformation').describe_stack_events(StackName=stack_id))
```

Comment:
doesn't this mean only `cloudformation` waiters are allowed / would work here? If so, why take the client as a parameter?